src/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyDependencyResolver.cs

```diff
@@ -232,6 +232,8 @@ private IEnumerable<LibraryNameVariation> DetermineLibraryNameVariations(string
             yield return new LibraryNameVariation(string.Empty, string.Empty);
 
             if (isRelativePath &&
+                libName.Contains('.') &&
```

Comment:
This part of logic is related to probing done by LoadLibrary. AssemblyDependencyResolver is feeding the name to File.Exists - this logic does not make sense for that case.